### PR TITLE
Learning rate explanation

### LINKED ISSUE #798: Correct explanation of learning_rate
The [explanation](https://inria.github.io/scikit-learn-mooc/python_scripts/ensemble_hyperparameters.html#histogram-gradient-boosting-decision-trees) of the `learning_rate` in the HistGradientBoostingClassifier does not agree with the scikit-learn [documentation](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html):

_"When fitting the residuals, we would like the tree to try to correct all possible errors or only a fraction of them. The learning-rate allows you to control this behaviour. A small learning-rate value would only correct the residuals of very few samples. If a large learning-rate is set (e.g., 1), we would fit the residuals of all samples."_

This explanation suggests that there is some selection of samples being made to which following trees make corrections. The documentation of scikit-learn says that the learning rate is a multiplicative factor, which I think is correct:

_"The learning rate, also known as shrinkage. This is used as a multiplicative factor for the leaves values. Use 1 for no shrinkage."_

### PR DESCRIPTION
**Pull request by fritshermans**
Fixes #798.

This PR corrects the explanation of the learning_rate parameter in GBDT models.

### CODE REVIEW

File: python_scripts/ensemble_hyperparameters.py [hunk 1]
```diff
@@ -159,15 +159,15 @@
...
-# control this behaviour. A small learning-rate value would only correct the
-# residuals of very few samples. If a large learning-rate is set (e.g., 1), we
-# would fit the residuals of all samples. So, with a very low learning-rate, we
-# would need more estimators to correct the overall error. However, a too large
-# learning-rate tends to obtain an overfitted ensemble, similar to having very
-# deep trees.
+# Finally, we have overlooked the impact of the `learning_rate` parameter
+# until now. This parameter controls how much each tree contributes to the
+# final prediction. A smaller learning-rate means the corrections of a new
+# tree results in small adjustments to the model prediction. When the
```
Comment (ArturoAmorQ) [inline python_scripts/ensemble_hyperparameters.py:165]:
```suggestion
# tree result in small adjustments to the model prediction. When the
```

File: python_scripts/ensemble_hyperparameters.py [hunk 2]
```diff
@@ -159,15 +159,15 @@
...
+# learning-rate is small, the model generally needs more trees to achieve good
+# performance. A higher learning-rate makes larger adjustments with each tree.
+# The model will need fewer trees and the model will train faster. The
```
Comment (ArturoAmorQ) [inline python_scripts/ensemble_hyperparameters.py:168]:
```suggestion
# performance. A higher learning-rate makes larger adjustments with each tree,
# which requires fewer trees and trains faster, at the risk of overfitting. The
```

File: python_scripts/ensemble_hyperparameters.py [hunk 3]
```diff
@@ -159,15 +159,15 @@
...
-# now. When fitting the residuals, we would like the tree to try to correct all
-# possible errors or only a fraction of them. The learning-rate allows you to
-# control this behaviour. A small learning-rate value would only correct the
-# residuals of very few samples. If a large learning-rate is set (e.g., 1), we
-# would fit the residuals of all samples. So, with a very low learning-rate, we
-# would need more estimators to correct the overall error. However, a too large
-# learning-rate tends to obtain an overfitted ensemble, similar to having very
-# deep trees.
+# Finally, we have overlooked the impact of the `learning_rate` parameter
+# until now. This parameter controls how much each tree contributes to the
```
Comment (ArturoAmorQ) [inline python_scripts/ensemble_hyperparameters.py:163]: You may need to re-format the length for this one:
```suggestion
# until now. This parameter controls how much each correction contributes to the
```

**Review by ArturoAmorQ — APPROVED** [review]
Thanks for addressing this issue @fritshermans. Just a couple of wording tweaks.

**Comment by ArturoAmorQ** [comment]
Thanks again @fritshermans, merging! 

On a side comment, hitting 800 in the github counter is quite a milestone :tada:
